packages/merkle-distributor/scripts/0_MerklePayoutIngestor.ts

```diff
@@ -0,0 +1,103 @@
+assert(isAddress(o.rewardToken), "Invalid rewardToken" + o.rewardToken);
+
+async function main() {
+  console.log("Running claims ingestion script 🎂");
+
+  // Add initial params to payout object and define the type
+  const outputObject: {
+    chainId: number;
+    rewardToken: string;
+    windowIndex: number;
+    totalRewardsDistributed: string;
+    recipients: {
+      [key: string]: { amount: string; metaData: Array<string>; accountIndex: number };
+    };
+  } = {
+    chainId: Number(o.chainId),
+    rewardToken: o.rewardToken,
+    windowIndex: Number(o.windowIndex),
+    totalRewardsDistributed: "0", // Initialized after all data read in. Sum of all rewards.
+    recipients: {} // Built up from the n input files defined in the params.
+  };
+
+  // Build up payouts for recipients for each input file.
+  for (let i = 0; i < o.input.length; i++) {
+    const inputFile = JSON.parse(fs.readFileSync(o.input[i], { encoding: "utf8" }));
+
+    // Fetch the recipients from the input file. The key donates the identifier within the input file. if this is null
+    // then the input file has no nested structures and only contains the recipients.
+    const recipients = o.key == "null" ? inputFile[o.key] : inputFile;
+    Object.keys(recipients).forEach((recipientAddress: string) => {
+      // Scale the amount by the number of decimals for that particular input.
+      const recipientAmountScaled = ConvertDecimals(0, o.decimals[i], Web3)(recipients[recipientAddress]);
+
+      // If the output file already contains information for this particular recipient, then append and add their rewards.
+      // Else, simply init the object with their values from the file. Note that accountIndex in both cases is set to 0.
+      // This will be set once the full data structure is built and ordered by payout amount.
+      if (outputObject.recipients[recipientAddress]) {
+        outputObject.recipients[recipientAddress] = {
+          amount: toBN(outputObject.recipients[recipientAddress].amount)
+            .add(recipientAmountScaled)
+            .toString(),
+          metaData: [...outputObject.recipients[recipientAddress].metaData, o.reason[i]],
+          accountIndex: 0
+        };
+      } else {
+        outputObject.recipients[recipientAddress] = {
+          amount: recipientAmountScaled.toString(),
+          metaData: [o.reason[i]],
+          accountIndex: 0
+        };
+      }
+    });
+  }
+
+  // Sort the outputs payment amount.
+  outputObject.recipients = Object.fromEntries(
+    Object.entries(outputObject.recipients).sort(([, a], [, b]) => (toBN(a.amount).gt(toBN(b.amount)) ? -1 : 0))
+  );
+
+  // Append an index to each recipient and count the totalRewardsDistributed.
+  let totalRewardsDistributed = toBN("0");
+  Object.keys(outputObject.recipients).forEach((recipientAddress: string, index: number) => {
+    outputObject.recipients[recipientAddress].accountIndex = index;
```

Comment:
I'd run a sanity check, before  writing the file to the desk, that all account indices are unique. If they are not unique it could mean `verifyProof` fails unexpectedly for some claims. You could try using `lodash`'s `uniqBy` method like so:

```js
const _ = require("lodash")
const uniqueIndices = _.uniqBy(outputObject.recipients, "accountIndex")
assert.equal(uniqueIndices.length, outputObject.recipients.length, "duplicate account indices")
```

Reply:
ye, that's a good check! will add it.

Reply:
using `[...new Set(Object.keys(outputObject.recipients))]` worked out much faster than lodash. using this. (this is es10 syntax I believe)